### DISCUSSION TITLE
Seperator doesn't look same

### DISCUSSION THREAD

**Discussion by SrivastavaArjit** [shadcn-ui/ui#3870] [General]
Hi folks,
I am using Shadcn for my Vite project and I am stuck with something.
So, I am using Shadcn seperator inside a map function but the **height of seperator doesn't look same.** Please have a look at the image.

![Screenshot 2024-05-30 212358](https://github.com/shadcn-ui/ui/assets/82926673/5c069e8f-eb9f-4475-a6a5-293a43538df1)

Below is the image of my map function: -
![Screenshot 2024-05-30 212710](https://github.com/shadcn-ui/ui/assets/82926673/84ffab3c-3eb1-4bcd-b085-55c47aee1ab7)

I have been stuck with this for more than 24 hours now and I have no idea why this is happening any explanation or solution will be greatly appreciated.

**Comment by adi907** [comment]
Facing the same issue
<img width="279" alt="image" src="https://github.com/shadcn-ui/ui/assets/76524120/2e0121ce-fbc2-457a-93a8-bbc9a4077993">

Both separators seem to be having different darkness eventhough no added class is applied to them

**Comment by mo-shahab** [comment]
I am having the same issue, why is shadcn so chaotic sometimes ?

**Comment by joeychrys** [comment]
Same problem :/. for me its every other separator. they appear to alternate between height when placed in succession. If I define the className="h-[1.5px]" for each one that is small it works.

**Comment by xf3227** [comment]
Same problem here. Here is the source styling code of the Seperator component.

```javascript
const Separator = React.forwardRef<
  React.ElementRef<typeof SeparatorPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof SeparatorPrimitive.Root>
>(
  (
    { className, orientation = "horizontal", decorative = true, ...props },
    ref
  ) => (
    <SeparatorPrimitive.Root
      ref={ref}
      decorative={decorative}
      orientation={orientation}
      className={cn(
        "shrink-0 bg-border",
        orientation === "horizontal" ? "h-[1px] w-full" : "h-full w-[1px]",
        className
      )}
      {...props}
    />
  )
)
```

We can see that it uses filled color (`bg-border` here) to render. Unfortunately, the size of a component is not absolute even with tailwind `flex-none` or `max-*` or `min-*`. This may explain why we see inconsistent rendering in different context.

Here is my solution, instead of using `Seperator`, we can simply use the following

```javascript
<span className="border h-4 mx-2 border-foreground"></span>  {/* Vertical */}
<span className="border w-4 my-2 border-foreground"></span>  {/* Horizontal */}
```

**The key idea is to utilize the border thickness** which is always consistently rendered to my best knowledge.

**Comment by Zk3619** [comment]
It happened when I was using solid-ui only my-1.5. But I found that it can be solved by adding h-0.3 or more height, but why?

**Reply by tkzero789** [reply]
Hey, I applied @xf3227's solution directly in the `Separator` component and it worked great for me. You should remove the `bg-` and use `border-t` or `border-b` for horizontal, and `border-r` or `border-l` for vertical.

**Comment by desmondinho** [comment]
I ran into the same thing one day (not even with shadcn) and also couldn’t figure out what was going on, since no extra styles were applied. Then I tried adjusting the page zoom and noticed that it “fixed” the problem. I came to the conclusion that it’s some resolution/pixel-related quirk, so I just stopped worrying about it.

**Comment by Spirit4449** [comment]
I am having the same issue
<img width="358" height="344" alt="image" src="https://github.com/user-attachments/assets/e923f89c-cba6-4838-87b7-a4d1e36b1142" />
`
<PopoverContent
        align="end"
        className="border-2 border-muted-foreground p-3"
      >
        <PopoverHeader>
          <PopoverTitle>Change Language</PopoverTitle>
          <PopoverDescription>
            Select your preferred language.
          </PopoverDescription>
        </PopoverHeader>

        <div className="flex flex-col gap-1">
          <LocaleOption
            locale="en"
            languageName={localeMeta.en.name}
            languageDescription={localeMeta.en.description}
            switchLocale={switchLocale}
          />
          <Separator />
          <LocaleOption
            locale="gu"
            languageName={localeMeta.gu.name}
            languageDescription={localeMeta.gu.description}
            switchLocale={switchLocale}
          />
          <Separator />

          <LocaleOption
            locale="hi"
            languageName={localeMeta.hi.name}
            languageDescription={localeMeta.hi.description}
            switchLocale={switchLocale}
          />
        </div>
      </PopoverContent>`
